src/main/java/com/google/sps/data/AvailabilityTimeSlot.java

```diff
@@ -0,0 +1,59 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package availability;
+
+/**
+ * An AvailabilityTimeSlot represents a table entry on the availability page that has a date and
+ * time (utcEncoding), a visual represenation of the time (timeString), and a class that is set
+ * based on whether or not the slot has already been selected in a previous visit to the page (green
+ * = selected = 'table-success').
+ */
+public class AvailabilityTimeSlot {
+  private String utcEncoding;
+  private String time;
+  private String date;
+  private String className;
```

Comment:
Mixing of "view" logic with how the servlet models the domain (i.e. this class) is a red flag.

I haven't scrutinized exactly how this is being used, but I just wanted to call that out. Mixing frontend and backend concerns results in weird coupling relationships, such as the backend being dependent on a certain arrangement of the elements, which can cause the application to "ossify" and be difficult to change.

Reply:
Handled